sd/internal/cache/cache.go

```diff
@@ -0,0 +1,104 @@
+package cache
```

Comment:
Oh! I realize I need to port over the atomic.Value changes here that were recently added.